cmd/fetchbranches/main.go

```diff
@@ -0,0 +1,113 @@
+// Copyright (c) 2017 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+func main() {
```

Comment:
This function is quite long - you might want to refactor it a little.

Another thing to think about is adding some unit tests for functions like `GetBranch()`, `GetPR()`, `GetRepository()` and `NewCI()`.

You might find some useful bits in https://github.com/clearcontainers/tests/blob/master/cmd/checkcommits/checkcommits_test.go as that too has to query Semaphore environment variables so I had to add tests for all sorts of interesting scenarios ;-)